src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java

```diff
@@ -95,20 +65,27 @@ public JWKSet getSigningKeys() {
     }
 
     @Override
-    public SignedJWT sign(JWTClaimsSet jwtClaimsSet) throws JOSEException {
-        var activeKeys = new ArrayList<>(getActiveSigningKeys());
-        Collections.shuffle(activeKeys, this.random); // Randomly shuffle our active keys, so we pick an arbitrary one first
+    @SneakyThrows
+    public SignedJWT sign(JWTClaimsSet jwtClaimsSet) {
+        var jwks = new ArrayList<>(getAllSigningKeys());
+
+        Collections.shuffle(jwks, this.random); // Randomly shuffle our active keys, so we pick an arbitrary one first
```

Comment:
```suggestion
        Collections.shuffle(jwks, this.random); // Randomly shuffle our keys, so we pick an arbitrary one first
```

This is no longer the list of only the active keys